tests/unit/backend/chalice/api_server/test_v1_collection.py

```diff
@@ -427,8 +430,143 @@ def test__list_collection__check_owner(self):
             self.assertIn(private_owned, ids)
             self.assertNotIn(private_not_owned, ids)
 
-    def test__delete_collection__OK(self):
-        path = furl(path="/dp/v1/collections/test_collection_id")
+
+class TestCollectionDeletion(BaseAuthAPITest, GenerateDataMixin):
+    def test_delete_collection__ok(self):
+        # Generate test collection
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        )
+        processing_status_1 = {"upload_status": UploadStatus.WAITING, "upload_progress": 0.0}
+        processing_status_2 = {"upload_status": UploadStatus.UPLOADED, "upload_progress": 100.0}
+
+        dataset_1 = self.generate_dataset(self.session, collection=collection, processing_status=processing_status_1)
+        dataset_2 = self.generate_dataset(self.session, collection=collection, processing_status=processing_status_2)
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        response = self.app.get(test_url.url, headers=headers)
+        response.raise_for_status()
+
+        body = json.loads(response.body)
+        dataset_ids = [dataset["id"] for dataset in body["datasets"]]
+        self.assertIn(dataset_1.id, dataset_ids)
+        self.assertIn(dataset_2.id, dataset_ids)
```

Comment:
Remove. We don't need to test if get collection works.

Reply:
When testing deletions I like to do an initial test to ensure it was created to make sure the test isnt passing due to the resource never being created

Reply:
but you can retrieve the dataset and collection and see that it's tombstoned using `get(tombstone=True)`, that will verify that it was both created and tombstoned. It looks like you're already doing that after the test. I still think this extra verify is superfluous